### PR TITLE
Checkout: Emergent Paywall custom css

### PR DESCRIPTION
**Pull request by yoavf**
## What is this
The Emergent Paywall (see #24675) loads in an iframe, so we can't adjust its internal styles with the primary Calypso stylesheet. This creates a separate CSS file to be linked from inside the iframe.

Needed for backend: D13666-code

### Screenshots

screen shot 2018-05-28 at 13 28 17

screen shot 2018-05-29 at 13 17 39

### Getting the paywall to show

Tweak `getCurrentUserPaymentMethods` in `client/state/selectors/get-current-user-payment-methods.js` by hardcoding `IN` as the countryCode for `paymentMethods.byCountry[ countryCode ] ||`.

### Seeing local css changes

To see local CSS changes you'll need to
make your browser rewrite those CSS URLs to the local version:
`https://wpcalypso.wordpress.com/calypso/emergent-paywall.css` => `http://calypso.localhost:3000/calypso/emergent-paywall.css`

You can accomplish this using a browser extension such as Switcharoo Redirector
for Chrome: https://chrome.google.com/webstore/detail/switcheroo-redirector/cnmciclhnghalnpfhhleggldniplelbg

An unfortunate side-effect is that the widget's iframe src is https, but this local CSS file is http. So most browsers will block the CSS by default to prevent mixed content. In order to see the custom CSS you will need to temporarily allow insecure content on the page. In Chrome for Mac you can issue this Terminal command:

`open -a Google\ Chrome --args --allow-running-insecure-content`
**Be sure to quit and restart your browser normally when you're done!**

### Backend
Apply D13666-code, which has some additinoal testing instructions

### Test payment info:
- *paytm*: 
  - Phone Number: 7777777777
  - OTP: 489871
- *Bank Transfer*
  - Bank: "AXIS Bank"
- Debit Card
  - Card no : 4012001037141112
  - Card expiry : 05/20
  - CVV : 123

### CODE REVIEW

File: assets/stylesheets/emergent-paywall.scss [hunk 1]
```diff
@@ -0,0 +1,687 @@
...
+}
+
+html,
+input,
+select,
+textarea {
+	font-family: $sans;
+}
+
+.body-hosted {
```
Comment (ramonjd) [inline assets/stylesheets/emergent-paywall.scss]: The `min-width` of  `562px` prevents the iframe from scaling in lower resolutions.

Something like this allows the content to scale:

```
.html-iframe body {
	display: block;
}

.body-hosted {
	min-width: auto; // we allow it to grow for embedded off site iframe
	width: 100%;
...

```

File: assets/stylesheets/emergent-paywall.scss [hunk 2]
```diff
@@ -0,0 +1,687 @@
...
+			width: 50%;
+			height: 100px;
+
+			.btn-tile {
+				text-align:center;
+				cursor: pointer;
+				padding: 10px;
+
+				img {
+					max-width: 220px;
```
Comment (ramonjd) [inline assets/stylesheets/emergent-paywall.scss]: At `220px`, the PayTM images overlap at lower viewport widths. Anything at or below `max-width: 180px;` seems to work.

File: assets/stylesheets/emergent-paywall.scss [hunk 3]
```diff
@@ -0,0 +1,687 @@
...
+	animation-name: rotate-spinner;
+}
+
+.busy-indicator, .less-busy-indicator {
+	border-top-color: $blue-medium;
+	width: 35px;
+	height: 35px;
+}
+
+.busy-indicator:after, .less-busy-indicator:after  {
```
Comment (ramonjd) [inline assets/stylesheets/emergent-paywall.scss]: Did stylelint complain about extra spaces and the other stuff? I see some gymnastics is required to overwrite the particularly bad markup, so no real drama.

File: assets/stylesheets/emergent-paywall.scss [hunk 4]
```diff
@@ -0,0 +1,687 @@
+// The Emergent Paywall loads in an iframe, so we can't adjust its internal styles
+// with the primary Calypso stylesheet. This file is bundled as a separate CSS file to be
+// linked from inside the iframe.
+//
+// Emergent requires that all custom CSS files exist at a public web address
+// So the CSS files Directlyhard-set to:
+//
+// Sandbox:    https://wpcalypso.wordpress.com/calypso/emergent-paywall.css
+// Production: https://wordpress.com/calypso/emegent-paywall.css
```
Comment (ramonjd) [inline assets/stylesheets/emergent-paywall.scss]: emergent

**Comment by ramonjd** [comment]
❤️ for the detailed test instructions!

File: assets/stylesheets/emergent-paywall.scss (hunk 4 above)
Comment (yoavf) [reply assets/stylesheets/emergent-paywall.scss]: 👍

**Comment by yoavf** [comment]
> ❤️ for the detailed test instructions!

props to @mattwondra - I got most of this from the [directly file](https://github.com/Automattic/wp-calypso/commits/master/assets/stylesheets/directly.scss) :)

**Comment by yoavf** [comment]
@ramonjd I did my best to fix the mobile issues (thanks for pointing them out). Should be OK now, I think :)

**Comment by yoavf** [comment]
I'll merge and test IE

**Comment by yoavf** [comment]
Tested IE11 - there are some issues wish initial payment method selection screen, but nothing too critical. I'll see if I can improve that. Not a blocker either way.

File: assets/stylesheets/emergent-paywall.scss [hunk 5]
```diff
@@ -0,0 +1,727 @@
...
+					}
+				}
+
+				.form-control {
+					margin: 0;
+					padding: 7px 14px;
+					color: $gray-dark;
+					font-size: 16px;
+					line-height: 1.5;
+					border: 1px solid #c8d7e1;
```
Comment (drw158) [inline assets/stylesheets/emergent-paywall.scss]: Any reason why hex values are being used for color? Ideally, there shouldn't be a need to use anything but the sass variable.

**Comment by drw158** [comment]
Just a couple things. The links at the bottom don't look like Calypso links, they don't appear to use standard colors. Also, these tabs are very verbose.

screen shot 2018-05-31 at 4 13 08 pm

But I know the original is also verbose.

Pretty confusing because debit card is in both tabs. If I had a magic wand, I think I'd name the tabs:

1. Card
2. Other

**Comment by yoavf** [comment]
@drw158 - thanks for jumping in!

In previous tests we've generally noticed that the more descriptive the tabs are, the better they are for conversion.

WRT to the duplication on credit and debit card, we're currently fixing that manually for users in India (who are the ones who will see this). 

This is how it looks:
screen shot 2018-06-18 at 15 16 44

**Pull request merged**